statsmodels/multivariate/cancorr.py

```diff
@@ -0,0 +1,185 @@
+        The canonical coefficients for exog
+
+    .. [1] http://numerical.recipes/whp/notes/CanonCorrBySVD.pdf
+    .. [2] http://www.csun.edu/~ata20315/psy524/docs/Psy524%20Lecture%208%20CC.pdf
+    .. [3] http://www.mathematica-journal.com/2014/06/canonical-correlation-analysis/
+    """
+    def __init__(self, endog, exog, missing='none', hasconst=None, **kwargs):
+        super(CanCorr, self).__init__(endog, exog, **kwargs)
+
+    def fit(self, tolerance=1e-8):
+        """Fit the model
+
+        A ValueError is raised if there are singular values smaller than the
+        tolerance. The treatment of singular arrays might change in future.
+
+        Parameters
+        ----------
+        tolerance : float
+            eigenvalue tolerance, values smaller than which is considered 0
+        Returns
+        -------
+        CanCorrResults instance
+
+        """
+        nobs, k_yvar = self.endog.shape
+        nobs, k_xvar = self.exog.shape
+        k = np.min([k_yvar, k_xvar])
+
+        x = np.array(self.exog)
+        x = x - x.mean(0)
+        y = np.array(self.endog)
+        y = y - y.mean(0)
+
+        ux, sx, vx = svd(x, 0)
+        # vx_ds = vx.T divided by sx
+        vx_ds = vx.T
+        mask = sx > tolerance
+        if mask.sum() < len(mask):
+            raise ValueError('exog is collinear.')
+        vx_ds[:, mask] /= sx[mask]
+        uy, sy, vy = svd(y, 0)
+        # vy_ds = vy.T divided by sy
+        vy_ds = vy.T
+        mask = sy > tolerance
+        if mask.sum() < len(mask):
+            raise ValueError('endog is collinear.')
+        vy_ds[:, mask] /= sy[mask]
+        u, s, v = svd(ux.T.dot(uy), 0)
+
+        # Correct any roundoff
+        self.cor_values= np.array([max(0, min(s[i], 1))
+                                   for i in range(len(s))])
+
+        self.x_cancoef = vx_ds.dot(u[:, :k])
+        self.y_cancoef = vy_ds.dot(v.T[:, :k])
+
+        return CanCorrResults(self)
+
+
+class CanCorrResults(object):
+    """
+    Canonical correlation results class
+
+    Parameters
+    ----------
+    fitted_cancorr : fitted CanCorr object
+
+    Attributes
+    -----------
+    stats : DataFrame
+        Contain statistical tests results for each canonical correlation
+    stats_mv : DataFrame
+        Contain the multivariate statistical tests results
+    """
+    def __init__(self, fitted_cancorr):
+        self.cor_values = fitted_cancorr.cor_values
+        self.x_cancoef = fitted_cancorr.x_cancoef
+        self.y_cancoef = fitted_cancorr.y_cancoef
+        self.nobs, self.k_yvar = fitted_cancorr.endog.shape
+        nobs, self.k_xvar = fitted_cancorr.exog.shape
+
+    def f_test(self):
+        """Approximate F test
+        Perform multivariate statistical tests of the hypothesis that
+        there is no canonical correlation between endog and exog.
+        For each canonical correlation, testing its significance based on
+        Wilks' lambda.
+
+        Returns
+        -------
+
```

Comment:
I didn't realize that this doesn't have a return.

I think this needs to return something instead of attaching all the results.

maybe not, CanCorrResults itself looks like a TestResults instance